services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy

```diff
@@ -410,87 +411,194 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
+        }
         islToBreak && antiflap.portUp(islToBreak.dstSwitch.dpId, islToBreak.dstPort)
         broughtDownPorts && broughtDownPorts.each { antiflap.portUp(it.switchId, it.portNo) }
         Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
             northbound.getAllLinks().each { assert it.state != FAILED }
         }
     }
 
+    @Tidy
+    def "Flow is not rerouted when switchUp event appear for a switch which is not related to the flow"() {
+        given: "Given a flow in DOWN status on neighboring switches"
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.paths.findAll { it.size() == 2 }.size() == 1
+        } ?: assumeTrue("No suiting switches found", false)
+
+        def flowPath = swP.paths.min { it.size() }
+        def altPath = swP.paths.findAll { it != flowPath }
+        def flow = flowHelperV2.randomFlow(swP)
+        flowHelperV2.addFlow(flow)
+        assert PathHelper.convert(northbound.getFlowPath(flow.flowId)) == flowPath
+
+        //All alternative paths for both flows are unavailable (bring ports down on the srcSwitch)
+        List<PathNode> broughtDownPorts = []
+        altPath.unique { it.first() }.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            antiflap.portDown(src.switchId, src.portNo)
+        }
+        Wrappers.wait(antiflapMin + WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        //move the flow to DOWN status
+        def islToBreak = pathHelper.getInvolvedIsls(flowPath).first()
+        antiflap.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        assert northbound.getFlowStatus(flow.flowId).status == FlowState.UP
+
+        when: "Generate switchUp event on switch which is not related to the flow"
+        def involvedSwitches = pathHelper.getInvolvedSwitches(flowPath)*.dpId
+        def switchToManipulate = topology.activeSwitches.find { !(it.dpId in involvedSwitches) }
+        lockKeeper.knockoutSwitch(switchToManipulate)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(switchToManipulate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+        def isSwitchActivated = false
+        lockKeeper.reviveSwitch(switchToManipulate)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(switchToManipulate.dpId).state == SwitchChangeType.ACTIVATED
+        }
+        isSwitchActivated = true
+
+        then: "Flow is not triggered for reroute due to switchUp event because switch is not related to the flow"
+        TimeUnit.SECONDS.sleep(rerouteDelay * 2) // it helps to be sure that the auto-reroute operation is completed
+        Wrappers.timedLoop(rerouteDelay) { // just in case
```

Comment:
I don't see any reason for this. prolonging test execution for another 2s

Reply:
I will remove it in further pr